packages/coreutils/src/time.ts

```diff
@@ -17,16 +17,25 @@ const UNITS: { name: Intl.RelativeTimeFormatUnit; milliseconds: number }[] = [
  * The namespace for date functions.
  */
 export namespace Time {
+  // Intl.RelativeTimeFormatStyle contains these, but we can't compile with it yet
```

Comment:
Can you expand on this comment?

Reply:
I had intended to use the type `Intl.RelativeTimeFormatStyle` to represent the different relative time format styles. This is part of the standard JavaScript library, but is not available in VS Code, and may not compile using the TypeScript version we use.

Reply:
What I meant is _which_ version of typescript is needed. From a quick check `typescript` `4.1.2` added  `es2020.intl` and I see `Intl.RelativeTimeFormatStyle` in https://github.com/microsoft/TypeScript/blob/release-4.1/src/lib/es2020.intl.d.ts whereas we use `5.1.6`, so I think we should be able to compile with it after adding `es2020.intl` to `compilerOptions` here:

https://github.com/jupyterlab/jupyterlab/blob/96c28b6107ffa7ff923819a3bec8b9be3b250925/packages/coreutils/tsconfig.json#L3

the same way as it is done for example here:

https://github.com/jupyterlab/jupyterlab/blob/96c28b6107ffa7ff923819a3bec8b9be3b250925/packages/ui-components/tsconfig.json#L3-L4

Reply:
The type `RelativeTimeFormatStyle` is part of the `Intl` namespace defined here, in `es2020.intl.d.ts`:

https://github.com/microsoft/TypeScript/blob/066773b99ba1867903182e3b1e3899c7b202b217/src/lib/es2020.intl.d.ts#L68-L73

I modified `tsconfig.json` in `tsconfig.json` in each of `packages/coreutils`, `packages/filebrowser`, and `packages/lsp-extension`, to include:

```
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2020.Intl"],
```

and `jlpm build` still fails with an error:

```
    ✖  @jupyterlab/lsp-extension:build:all
       > @jupyterlab/lsp-extension@4.2.0-beta.0 build
       > tsc -b
       
       ../coreutils/lib/time.d.ts(12,62): error TS2694: Namespace 'Intl' has no exported member 'RelativeTimeFormatStyle'.
       npm ERR! Lifecycle script `build` failed with error: 
       npm ERR! Error: command failed 
       npm ERR!   in workspace: @jupyterlab/lsp-extension@4.2.0-beta.0 
       npm ERR!   at location: /Users/jweill/git/jupyterlab/packages/lsp-extension 
```

The problem persists after running `jlpm install`, then `jlpm build` again.

Reply:
Yes, I see it. So this is because other packages such as `translation`, `statusbar`, etc also need it because they also depend on coreutils (you can debug this by compiling `lsp-extension` with `-v` flag). The simple solution would be to set it in tsconfigbase:

https://github.com/jupyterlab/jupyterlab/blob/6152e3a5f94d728ccd2727d6d8413f12ddafb669/tsconfigbase.json#L3

But this might be more disruptive so I think we do not have to do this.

At some point we should update our target to ES2020.

For now I would suggest to make the comment more specific/easier to discover when searching for ES2018/2020:

```suggestion
  // Intl.RelativeTimeFormatStyle contains but it requires `ES2020.Intl` (as of writing we use `ES2018` target)
```

so this is easier to find when we will be making the transition.